homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -55,12 +55,27 @@ class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
-    _comportslist = None
-    _defaultcomport = None
 
     def __init__(self):
         """Initialise the config flow."""
         self.config = None
+        self._comportslist = None
+        self._defaultcomport = None
+
+    def scan_comports(self):
+        """Find and store available com ports for the GUI dropdown."""
+        comports = serial.tools.list_ports.comports(include_links=True)
```

Comment:
This likely does I/O so it should probably be run in the executor

Reply:
Inside the library, it is [searching the filesystem](https://github.com/pyserial/pyserial/blob/9e3b4f364aa50bd62d3044ed0d5221bd42a1c1f4/serial/tools/list_ports_linux.py#L91) or making a [whole load of winapi calls](https://github.com/pyserial/pyserial/blob/9e3b4f364aa50bd62d3044ed0d5221bd42a1c1f4/serial/tools/list_ports_windows.py#L238).

Either way, I've put it into an executor block - I hope that was what you meant. :)